proxy.go

```diff
@@ -223,14 +221,13 @@ func (rp *reverseProxy) serveFromCache(s *scope, srw *statResponseWriter, req *h
 		return
 	}
 
+	// TODO: add tes case to verify that body remains unchanged
```

Comment:
Just implement this TODO - create compressed request, proxy it to fake clickhouse server and make sure the server receives the same compressed request.